### PR TITLE
[ONNX] Reduce* operations

### PR DESCRIPTION
**Pull request by arogowie-intel**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace reduction
+        {
+            namespace detail
+            {
+                inline std::vector<std::size_t> get_reduction_axes(const Node& node)
```
Comment (tsocha) [inline src/ngraph/frontend/onnx_import/utils/reduction.hpp]: Consider use of ngraph::AxisVector as return type.

File: src/ngraph/frontend/onnx_import/op/reduce.hpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
+
+#include "ngraph/node_vector.hpp"
+#include "ngraph/op/abs.hpp"
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/reduce.hpp]: Please consider moving the implementation into a `.cpp` file, so you won't have to include all of these headers in this header, which is included by the `ops_bridge`.

File: src/ngraph/frontend/onnx_import/utils/common.hpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                    ///
+                    template <typename ForwardIterator, typename T>
+                    void fill_monotonic_range(ForwardIterator first,
+                                              ForwardIterator last,
+                                              T init_value,
+                                              T step)
+                    {
+                        while (first != last)
+                        {
+                            *first++ = init_value;
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/utils/common.hpp]: I think regular `for()` is better, please notice post increment requires additional memory to hold previous state
```
for (;first != last; ++first, init_value += step)
{
    *first = init_value;
}
```

File: src/ngraph/frontend/onnx_import/utils/reshape.cpp [hunk 4]
```diff
@@ -59,9 +60,8 @@ namespace ngraph
 
             AxisVector get_default_axis_vector(std::size_t data_shape_size, std::size_t start_value)
             {
-                AxisVector axis_vector(data_shape_size);
-                std::iota(std::begin(axis_vector), std::end(axis_vector), start_value);
-                return axis_vector;
+                return AxisVector{onnx_import::common::get_monotonic_range<std::size_t>(
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/utils/reshape.cpp]: `common::get_monotonic_range<>` unless this piece of code is not part of `onnx_import` namespace.

Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/utils/reshape.cpp]: and `return {...}`, the `AxisVector` is not required.

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                    return op_node;
+                }
+
+                auto output_shape = data_shape;
+                // flatten reduced axes
+                for (const auto& idx : reduction_axes)
+                {
+                    output_shape.at(idx) = 1;
+                }
+                auto reshaped_op_node = std::make_shared<ngraph::op::Reshape>(
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/utils/reduction.hpp]: `return std::make_shared<ngraph::op::Reshape>(...);` the variable `reshape_op_node` is not required.

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                inline AxisSet get_reduction_axes(const Node& node)
+                {
+                    auto reduction_axes =
+                        node.get_attribute_value<std::vector<std::size_t>>("axes", {});
+                    if (reduction_axes.empty())
+                    {
+                        reduction_axes = onnx_import::common::get_monotonic_range<std::size_t>(
+                            node.get_ng_inputs().at(0)->get_shape().size());
+                    }
+                    return AxisSet{reduction_axes};
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/utils/reduction.hpp]: `AxisSet` is not required unless it is `explicit` then it is required.

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp (hunk 5 above)
Comment (postrational) [reply src/ngraph/frontend/onnx_import/utils/reduction.hpp]: The variable `reshape_op_node` makes the code more readable. Unless we're making an unnecessary copy here, let's keep it.

File: src/ngraph/frontend/onnx_import/utils/reshape.cpp (hunk 4 above)
Comment (postrational) [reply src/ngraph/frontend/onnx_import/utils/reshape.cpp]: Does adding `AxisVector` hurt anything? If not then, it makes the code more explicit and less magical...

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp (hunk 6 above)
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/utils/reduction.hpp]: I think that this code is more readable.

File: src/ngraph/frontend/onnx_import/utils/reshape.cpp (hunk 4 above)
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/utils/reshape.cpp]: I think that returning `AxisVector{...}` is more readable.

File: src/ngraph/frontend/onnx_import/utils/reduction.hpp (hunk 5 above)
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/utils/reduction.hpp]: The `return std::make_shared<ngraph::op::Reshape>()` clearly states that the function returns node which does reshape op.

**Pull request merged**